polaris-react/scripts/utilities/getCustomPropertyNames.js

```diff
@@ -25,4 +25,24 @@ const getCustomPropertyNames = () => {
   return polarisTokenCustomProperties;
 };
 
-module.exports = getCustomPropertyNames;
+const getGroupedCustomPropertyNames = () => {
+  const groupedTokens = {};
+  fs.readdirSync(tokenGroupsDir).forEach((file) => {
+    const tokenGroup = require(path.join(tokenGroupsDir, file));
+    let filename = file;
+
+    if (file.includes('color')) {
+      filename = 'color';
+    }
+
+    const tokenGroupCustomProperties = Object.keys(tokenGroup).map(
+      (token) => `--p-${token}`,
+    );
+
+    groupedTokens[filename.replace('.json', '')] = tokenGroupCustomProperties;
+  });
+
+  return groupedTokens;
+};
```

Comment:
```suggestion
const getGroupedCustomPropertyNames = () => {
  return Object.fromEntries(
    fs.readdirSync(tokenGroupsDir).map((fileName) => {
      const tokenGroup = require(path.join(tokenGroupsDir, fileName));
      
      let tokenGroupName = path.basename(fileName, '.json');

      if (fileName.startsWith('color.')) {
        tokenGroupName = 'color';
      }

      const customPropertyNames = Object.keys(tokenGroup).map(
        (token) => `--p-${token}`,
      );

      return [tokenGroupName, customPropertyNames]
    }),
  )
};
```

Optional suggestion. Refactored to use `Object.fromEntries` and `path.basename` API to get retrieve the `tokenGroupName`.

Reply:
i like this 💯 and had no idea this was a thing! do you mind creating a separate PR against `main` after we merge this PR so that we can unblock this current stream of work quickly?